hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java

```diff
@@ -0,0 +1,252 @@
+                                Tuple.of(
+                                        NEW_NAME,
+                                        NEW_SYMBOL,
+                                        asHeadlongAddress(SENDER_ADDRESS.toByteArray()),
+                                        "memo",
+                                        true,
+                                        1000L,
+                                        false,
+                                        // TokenKey
+                                        new Tuple[] {},
+                                        // Expiry
+                                        Tuple.of(123456L, asAddress(""), 0L)))
+                        .array()),
+                output -> assertEquals(
+                        Bytes.wrap(
+                                ReturnTypes.encodedRc(INVALID_EXPIRATION_TIME).array()),
+                        output));
+    }
+
+    @Override
+    protected void assertExpectedTokens(@NonNull ReadableKVState<TokenID, Token> tokens) {
+        final var erc20Token = tokens.get(ERC20_TOKEN_ID);
+        assertNotNull(erc20Token);
+        assertEquals(NEW_NAME, erc20Token.name());
+        assertEquals(NEW_SYMBOL, erc20Token.symbol());
+    }
+
+    @Override
+    protected Map<ProtoBytes, AccountID> initialAliases() {
+        final var aliases = new HashMap<ProtoBytes, AccountID>();
+        aliases.put(ProtoBytes.newBuilder().value(SENDER_ADDRESS).build(), SENDER_ID);
+        return aliases;
+    }
+
+    @Override
+    protected Map<TokenID, Token> initialTokens() {
+        final var tokens = new HashMap<TokenID, Token>();
+        tokens.put(
+                ERC20_TOKEN_ID,
+                Token.newBuilder()
+                        .tokenId(ERC20_TOKEN_ID)
+                        .treasuryAccountId(SENDER_ID)
+                        .tokenType(TokenType.FUNGIBLE_COMMON)
+                        .supplyKey(AN_ED25519_KEY)
+                        .adminKey(SENDER_CONTRACT_ID_KEY)
+                        .autoRenewAccountId(SENDER_ID)
+                        .build());
+        return tokens;
+    }
+
+    @Override
+    protected Map<EntityIDPair, TokenRelation> initialTokenRelationships() {
+        final var tokenRelationships = new HashMap<EntityIDPair, TokenRelation>();
+        addErc20Relation(tokenRelationships, SENDER_ID, 0);
+        return tokenRelationships;
+    }
+
+    @Override
+    protected Map<AccountID, Account> initialAccounts() {
+        final var accounts = new HashMap<AccountID, Account>();
+        accounts.put(
+                SENDER_ID,
+                Account.newBuilder()
+                        .accountId(OWNER_ID)
+                        .alias(SENDER_ADDRESS)
+                        .key(SENDER_CONTRACT_ID_KEY)
+                        .smartContract(true)
+                        .build());
+        return accounts;
+    }
+
+    private static Address asAddress(String address) {
```

Comment:
Can we move `asAddress` method to a util file specially for X tests, something like `XUtils`? cc: @tinker-michaelj

Reply:
Seems reasonable to me!

Reply:
@agadzhalov created a separate issue (#9179) for this